README.md

```diff
@@ -10,213 +10,302 @@
 ```
 
-On Windows you should run `nmake` instead of `make`
+_Windows_
+```bat
+set GST_PLUGIN_PATH=%CD%\.
+```
+
+Running the following command should now display information on the kvssink plugin:
+```
+gst-inspect-1.0 kvssink
+```
+If the build failed or GST_PLUGIN_PATH is not properly set, you may instead see the following output:<br>
+`No such element or plugin kvssink`
 
-In your build directory you will now have shared objects for all the targets you have selected.
+After building the SDK, loading kvssink into the GStreamer plugin path, and setting a region, the sample executables, which are located in the `build` directory, can be run. For example:
+```
+./kvssink_gstreamer_sample MyStreamName
+```
+To stream from file with this sample, you can download a test video file and append the file name after the stream name in the sample run command:
+```
+wget https://awsj-iot-handson.s3-ap-northeast-1.amazonaws.com/kvs-workshop/sample.mp4
+```
+```
+./kvssink_gstreamer_sample MyStreamName ./sample.mp4
+```
 
-### Installing the library
-If the library needs to be installed, run `make install`. This will install in default directory based on system. To install in another directory, run `cmake` with the `-DCMAKE_INSTALL_PREFIX` option with the desired directory before running `make install`
-## Run
-### GStreamer Plugin (kvssink)
+<br>
 
-#### Loading Element
-The GStreamer plugin is located in your `build` directory.
+**Running the kvssink Intermittent Sample**
 
-To load this plugin set the following environment variables. This should be run from the root of the repo, NOT the `build` directory.
+The intermittent kvssink sample will stream video for 20 seconds, then pause for 40 seconds, and repeat until an interrupt signal is received. To manually adjust the streaming and paused intervals, you can change the `KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS` and `KVS_INTERMITTENT_PAUSED_INTERVAL_SECONDS` values in the *kvssink_intermittent_sample.cpp* file.
 
 ```bash
-export GST_PLUGIN_PATH=`pwd`/build
-export LD_LIBRARY_PATH=`pwd`/open-source/local/lib
+./kvssink_intermittent_sample <stream-name> <testsrc or devicesrc (optional)>
+```
+Setting the source to `testsrc` will use [videotestsrc](https://gstreamer.freedesktop.org/documentation/videotestsrc/?gi-language=c) and to `devicesrc` will use [autovideosrc](https://gstreamer.freedesktop.org/documentation/autodetect/autovideosrc.html?gi-language=c). By default, kvssink uses "DEFAULT_STREAM" as the stream name, and the sample uses videotestsrc as the source. If a KVS stream with the provided or default name does not exist, the stream will automatically be created.
+
+<br>
+
+#### Running Samples in offline mode
+By default, the samples run in near-real-time mode. To set offline mode, set streamInfo.streamCaps.streamingType to `STREAMING_TYPE_OFFLINE`, where, `streamInfo` is of type `StreamInfo`, `streamCaps` is of type `StreamCaps` and `streamingType` is of type `STREAMING_TYPE`.
```

Comment:
I think this is confusing, you can explain this a bit better

Reply:
This I just copied over from the existing ReadMe, but looking at it now, we don't expose a StreamInfo object in the samples. Instead, the `streaming_type` variable should be modified to change the streaming type. However, I don't see why this would ever need to be done if the sample already changes it to offline mode when streaming from file. I'll remove this section entirely if you agree.

Reply:
Yeah just remove it. It's already explained in docs/structures.md